metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -349,6 +357,16 @@ public void replay(PartitionChangeRecord record) {
         log.debug("Applied ISR change record: {}", record.toString());
     }
 
+    public void replay(RemoveTopicRecord record) {
+        TopicControlInfo topic = topics.remove(record.topicId());
+        if (topic == null) {
+            throw new RuntimeException("Can't find topic with ID " + record.topicId() +
+                " to remove.");
+        }
+        topicsByName.remove(topic.name);
+        log.info("Removed topic {} with ID {}.", topic.name, record.topicId());
+    }
```

Comment:
I guess we haven't hooked up the logic to trigger the deletion of the replicas of the deleted topic in the broker?

Reply:
We haven't hooked that up yet, correct.  But that logic is in `BrokerMetadataListener`. It would probably be better to have a separate PR for that.